users/models.py

```diff
@@ -153,6 +159,28 @@ def is_employee(self) -> bool:
     def get_reports_created(self) -> QuerySet:
         return self.report_set.select_related("task_activities").order_by("-date", "project__name")
 
+    def get_projects_work_percentage(self, from_date: Optional[date] = None, to_date: Optional[date] = None) -> dict:
+        """
+        Returns dict where keys are Project objects and values are percentage amount of work
+        by this user between given dates.
+        """
+        if from_date is None:
+            from_date = timezone.now().date().replace(day=1)
+        if to_date is None:
+            to_date = timezone.now().date()
+
+        work_hours_per_project = self.projects.filter(report__date__range=[from_date, to_date]).annotate(
+            work_hours_sum=Coalesce(Sum("report__work_hours", filter=Q(report__author=self)), timezone.timedelta())
+        )
+        work_hours_sum = work_hours_per_project.aggregate(
+            sum_work_hours_sum=Coalesce(Sum("work_hours_sum"), timezone.timedelta())
+        )["sum_work_hours_sum"]
+
+        return {
+            project: ((project.work_hours_sum / work_hours_sum) * 100 if work_hours_sum.total_seconds() > 0 else 0)
```

Comment:
But the if is to avoid `ZeroDivisionError`

```
>>> import datetime
>>> datetime.timedelta(days=1) / datetime.timedelta()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ZeroDivisionError: division by zero
```

Reply:
Again, my mistake ;)